drivers/entropy/entropy_stm32.c

```diff
@@ -224,22 +226,36 @@ static void configure_rng(void)
 	cur_htcr = LL_RNG_GetHealthConfig(rng);
 #endif /* health_test_config */
 
-	if (cur_nist_cfg != desired_nist_cfg || cur_htcr != desired_htcr) {
+	if (cur_nist_cfg != desired_nist_cfg || cur_htcr != desired_htcr ||
+	    cur_nscr != desired_nscr) {
 		stm32_reg_modify_bits(&rng->CR, cur_nist_cfg, desired_nist_cfg | RNG_CR_CONDRST);
 
+#if defined(CONFIG_SOC_SERIES_STM32L4X)
```

Comment:
To be sure we clean up this mess in due time:
```suggestion
/*
 * TODO: this is needed only due to mismatch of STM32L4 HAL with others.
 * Remove this special case once STM32CubeL4 v1.18.2 or later has been
 * integrated in hal_stm32 (see also ST Internal Reference 207828).
 */
#if defined(CONFIG_SOC_SERIES_STM32L4X)
```